modules/text2vec-cohere/neartext/searcher.go

```diff
@@ -0,0 +1,140 @@
+	CombineVectors(vectors [][]float32) []float32
+}
+
+func (s *Searcher) VectorSearches() map[string]modulecapabilities.VectorForParams {
+	vectorSearches := map[string]modulecapabilities.VectorForParams{}
+	vectorSearches["nearText"] = s.vectorForNearTextParam
+	return vectorSearches
+}
+
+func (s *Searcher) vectorForNearTextParam(ctx context.Context, params interface{}, className string,
+	findVectorFn modulecapabilities.FindVectorFn,
+	cfg moduletools.ClassConfig,
+) ([]float32, error) {
+	return s.vectorFromNearTextParam(ctx, params.(*NearTextParams), className, findVectorFn, cfg)
+}
+
+func (s *Searcher) vectorFromNearTextParam(ctx context.Context,
+	params *NearTextParams, className string, findVectorFn modulecapabilities.FindVectorFn,
+	cfg moduletools.ClassConfig,
+) ([]float32, error) {
+	// it is safe to call NewClassSettings even knowing that cfg can be nil, it
+	// is to built to work with all defaults in the case of a nil-config, see
+	// vectorizer/class_settings_test.go for details.
+	settings := localvectorizer.NewClassSettings(cfg)
+	vector, err := s.vectorizer.Texts(ctx, params.Values, settings)
+	if err != nil {
+		return nil, errors.Errorf("vectorize keywords: %v", err)
+	}
+
+	moveTo := params.MoveTo
+	if moveTo.Force > 0 && (len(moveTo.Values) > 0 || len(moveTo.Objects) > 0) {
+		moveToVector, err := s.vectorFromValuesAndObjects(ctx, moveTo.Values,
+			moveTo.Objects, className, findVectorFn, settings)
+		if err != nil {
+			return nil, errors.Errorf("vectorize move to: %v", err)
+		}
+
+		afterMoveTo, err := s.vectorizer.MoveTo(vector, moveToVector, moveTo.Force)
+		if err != nil {
+			return nil, err
+		}
+		vector = afterMoveTo
+	}
+
+	moveAway := params.MoveAwayFrom
+	if moveAway.Force > 0 && (len(moveAway.Values) > 0 || len(moveAway.Objects) > 0) {
+		moveAwayVector, err := s.vectorFromValuesAndObjects(ctx, moveAway.Values,
+			moveAway.Objects, className, findVectorFn, settings)
+		if err != nil {
+			return nil, errors.Errorf("vectorize move away from: %v", err)
+		}
+
+		afterMoveFrom, err := s.vectorizer.MoveAwayFrom(vector, moveAwayVector, moveAway.Force)
+		if err != nil {
+			return nil, err
+		}
+		vector = afterMoveFrom
+	}
+
+	return vector, nil
+}
+
+func (s *Searcher) vectorFromValuesAndObjects(ctx context.Context,
+	values []string, objects []ObjectMove,
+	className string,
+	findVectorFn modulecapabilities.FindVectorFn,
+	settings localvectorizer.ClassSettings,
+) ([]float32, error) {
+	var objectVectors [][]float32
+
+	if len(values) > 0 {
+		moveToVector, err := s.vectorizer.Texts(ctx, values, settings)
+		if err != nil {
+			return nil, errors.Errorf("vectorize move to: %v", err)
+		}
+		objectVectors = append(objectVectors, moveToVector)
+	}
+
+	if len(objects) > 0 {
+		var id strfmt.UUID
+		for _, obj := range objects {
+			if len(obj.ID) > 0 {
+				id = strfmt.UUID(obj.ID)
+			}
+			if len(obj.Beacon) > 0 {
+				ref, err := crossref.Parse(obj.Beacon)
```

Comment:
parsed `ref` can have a `Class` information in it, so if `ref.Class != ""` please pass this information to `findVectorFn` method like here the `id` is passed

Reply:
Resolved